apm-sniffer/optional-plugins/optional-spring-plugins/spring-webflux-5.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/spring/webflux/v5/reactive/DispatcherHandlerHandleMethodInterceptor.java

```diff
@@ -16,62 +16,93 @@
 import org.springframework.web.server.ServerWebExchangeDecorator;
 import org.springframework.web.server.adapter.DefaultServerWebExchange;
+import reactor.core.publisher.Mono;
 
-import java.lang.reflect.Method;
-import java.util.List;
-
+/**
+ * @author Born
+ */
 public class DispatcherHandlerHandleMethodInterceptor implements InstanceMethodsAroundInterceptor {
 
-    private static final String WIP_OPERATION_NAME = "WEBFLUX.handle";
+    private ContextManagerExtendService extendService = ServiceManager.INSTANCE.findService(ContextManagerExtendService.class);
 
     @Override
     public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                              MethodInterceptResult result) throws Throwable {
-        EnhancedInstance instance = DispatcherHandlerHandleMethodInterceptor.getInstance(allArguments[0]);
-        if (instance != null) {
-            ContextCarrier contextCarrier = new ContextCarrier();
-            CarrierItem next = contextCarrier.items();
-            ServerWebExchange exchange = (ServerWebExchange) allArguments[0];
-            HttpHeaders headers = exchange.getRequest().getHeaders();
-            while (next.hasNext()) {
-                next = next.next();
-                List<String> header = headers.get(next.getHeadKey());
-                if (header != null && header.size() > 0) {
-                    next.setHeadValue(header.get(0));
-                }
-            }
-            AbstractSpan span = ContextManager.createEntrySpan(WIP_OPERATION_NAME, contextCarrier);
-            span.setComponent(ComponentsDefine.SPRING_WEBFLUX);
-            SpanLayer.asHttp(span);
-            Tags.URL.set(span, exchange.getRequest().getURI().toString());
-            instance.setSkyWalkingDynamicField(span);
-        }
+
     }
 
     @Override
     public Object afterMethod(EnhancedInstance objInst, Method method, Object[] allArguments, Class<?>[] argumentsTypes,
                               Object ret) throws Throwable {
-        return ret;
+        EnhancedInstance instance = getInstance(allArguments[0]);
+
+        ServerWebExchange exchange = (ServerWebExchange) allArguments[0];
+
+        ContextCarrier carrier = new ContextCarrier();
+        CarrierItem next = carrier.items();
+        HttpHeaders headers = exchange.getRequest().getHeaders();
+        while (next.hasNext()) {
+            next = next.next();
+            List<String> header = headers.get(next.getHeadKey());
+            if (header != null && header.size() > 0) {
+                next.setHeadValue(header.get(0));
+            }
+        }
+
+        String pathStr = exchange.getRequest().getPath().value();
+        String methodStr = exchange.getRequest().getMethodValue();
+        String operationName = methodStr + ":" + pathStr;
+
+        AbstractTracerContext context = extendService.createTraceContext(operationName, false);
```

Comment:
Use ContextManager is not enough? extendService.createTraceContext API  is not very suggested... FYI @wu-sheng

Reply:
Only `ContextManager` is open API, good catch. Please change to the right API.

Reply:
ContextManager Internally use ThreadLocal to save Context.  In this scenario, ThreadLocal cannot be used

Reply:
Let's talk about this more. If you use `extendService`, who will manage the AbstractTracerContext?

Reply:
And why ThreadLocal should be avoided.

Reply:
Please say more detail , You want control Context at skyWalkingDynamicField? but when createEntry span ThreadLocal will used.

Reply:
The reason we suggest and open ContextManager, because the plugin should be able to be aware of other plugins in the same context. Once you created the Context and managed by yourself, span created by other spans can't be linked to the Context you created. Because they can't find your Context.